content/doc/book/system-administration/authenticating-scripted-clients.adoc

```diff
@@ -0,0 +1,378 @@
+
+Note that Jenkins does not do any authorization negotiation. 
+i.e. it immediately returns a 403 (Forbidden) response instead
+of a 401 (Unauthorized) response, so make sure to send the authentication
+information from the first request (aka "preemptive authentication").
+
+== Groovy script using cdancy/jenkins-rest
+
+The https://github.com/cdancy/jenkins-rest[cdancy/jenkins-rest client]
+greatly simplifies REST API access. 
+The following Groovy code shows how to authenticate to Jenkins and get some system info:
+
+[source,groovy]
+----
+@Grab(group='com.cdancy', module='jenkins-rest', version='0.0.18')
+
+import com.cdancy.jenkins.rest.JenkinsClient
+
+JenkinsClient client = JenkinsClient.builder()
+    .endPoint("http://127.0.0.1:8080") // Optional. Defaults to http://127.0.0.1:8080
+    .credentials("user:apiToken") // Optional.
+    .build()
+
+println(client.api().systemApi().systemInfo())
+----
+
+For additional information, see the
+https://github.com/cdancy/jenkins-rest/wiki[cdancy/jenkins-rest wiki].
+
+[[Authenticatingscriptedclients-Groovyscriptusingcommons-httpclient]]
+== Groovy script using commons-httpclient
+
+In a groovy script, this could look something like this (using
+commons-httpclient):
+
+[source,groovy]
+----
+import org.apache.commons.httpclient.*
+
+import org.apache.commons.httpclient.auth.*
+import org.apache.commons.httpclient.methods.*
+
+@Grab(group='commons-httpclient', module='commons-httpclient', version='3.1')
+void createNewJenkinsProject() {
+
+  def server = "server"
+  def jenkinsHost = "https://${server}/jenkins/"
+  def projectName = "TEST"
+  def configurationFile = "config.xml"
+
+  def username = "username"
+  def apiToken = "apiToken"
+
+  def client = new HttpClient()
+  client.state.setCredentials(
+    new AuthScope( server, 443, "realm"),
+    new UsernamePasswordCredentials( username, apiToken )
+  )
+
+  // Jenkins does not do any authentication negotiation,
+  // i.e. it does not return a 401 (Unauthorized)
+  // but immediately a 403 (Forbidden)
+  client.params.authenticationPreemptive = true
+
+  def post = new PostMethod( "${jenkinsHost}/createItem?name=${projectName}" )
+  post.doAuthentication = true
+
+  File input = new File(configurationFile);
+  RequestEntity entity = new FileRequestEntity(input, "text/xml; charset=UTF-8");
+  post.setRequestEntity(entity);
+  try {
+    int result = client.executeMethod(post)
+    println "Return code: ${result}"
+    post.responseHeaders.each{ println it.toString().trim() }
+    println post.getResponseBodyAsString()
+  } finally {
+    post.releaseConnection()
+  }
+}
+createNewJenkinsProject()
+----
+
+
+NOTE: If you are using wget 1.11 against Jenkins version 1.586 and above
+with the
+https://issues.jenkins-ci.org/browse/JENKINS-25169[JENKINS-25169] fix,
+you might need to use the following options:
+
+[source,sh]
+----
+wget --auth-no-challenge \
+    --http-user=user --http-password=apiToken \
+    --secure-protocol=TLSv1 \
```

Comment:
tls v1 is deprecated and not well supported anymore, let the client and server negotiate
```suggestion
```